src/components/audio-params.js

```diff
@@ -298,36 +307,42 @@ AFRAME.registerComponent("audio-params", {
 
   updateDistances() {
     this.el.sceneEl.audioListener.getWorldPosition(this.listenerPos);
-    this.data.distance = this.data.position.distanceTo(this.listenerPos);
-    this.data.squaredDistance = this.data.position.distanceToSquared(this.listenerPos);
+    this.el.setAttribute("audio-params", {
+      distance: this.data.position.distanceTo(this.listenerPos),
+      squaredDistance: this.data.position.distanceToSquared(this.listenerPos)
+    });
   },
 
   updateAttenuation() {
-    this.data.attenuation = distanceModels[this.data.distanceModel](
-      this.data.distance,
-      this.data.rolloffFactor,
-      this.data.refDistance,
-      this.data.maxDistance
-    );
+    this.el.setAttribute("audio-params", {
+      attenuation: distanceModels[this.data.distanceModel](
+        this.data.distance,
+        this.data.rolloffFactor,
+        this.data.refDistance,
+        this.data.maxDistance
+      )
+    });
   },
 
   clipGain(gain) {
     if (!this.data.isClipped) {
       const audio = this.getAudio();
-      this.data.isClipped = true;
-      this.data.preClipGain = this.data.gain;
-      this.data.gain = gain;
-      this.data.gain = Math.max(0.001, this.data.gain);
+      this.el.setAttribute("audio-params", {
+        isClipped: true,
+        preClipGain: this.data.gain,
+        gain: Math.max(0.001, gain)
+      });
       audio.gain.gain.exponentialRampToValueAtTime(this.data.gain, audio.context.currentTime + MUTE_DELAY_SECS);
```

Comment:
We can remove this line because `gain` is applied in `update`

Reply:
Same as before, we were updating the gain twice, once on the audio node gain and another one in the gain filter.